src/PolykeyClient.ts

```diff
@@ -57,20 +54,17 @@ class PolykeyClient {
         logger: logger.getChild(Session.name),
         fresh,
       }));
-    grpcClient =
-      grpcClient ??
-      (await GRPCClientClient.createGRPCClientClient({
-        nodeId,
-        host,
-        port,
-        tlsConfig: { keyPrivatePem: undefined, certChainPem: undefined },
-        session,
-        timer,
-        logger: logger.getChild(GRPCClientClient.name),
-      }));
+    const rpcClient = await RPCClient.createRPCClient<M>({
+      manifest,
+      streamFactory,
+      middleware: middlewareUtils.defaultClientMiddlewareWrapper(
+        authMiddleware.authenticationMiddlewareClient(session),
+      ),
+      logger: logger.getChild(RPCClient.name),
+    });
```

Comment:
Change this to be DIed optionally like we did with `grpcClient`. The DI enables mock testing.

Reply:
I'm thinking about how to set the manifest or DI the `RPCClient`. Technically the options are mutually exclusive. Ideally this would be enforced with the type.

I can think of two options. the `manifest` and `rpcClient` are both optional but creation will fail if if neither are provided. Providing both would default to the DI-ed `rpcClient`.

The other option is to have the `manifest` be the `RPCClient|ClientManifest`, if the manifest is provided then we create the `rpcClient`. 

I'm not sure which one is preferred here.

Reply:
I'm going with option 2, it's much cleaner.

Reply:
This is fixed.